x-pack/plugins/triggers_actions_ui/public/common/lib/data_apis.ts

```diff
@@ -61,11 +68,44 @@ export const getSavedObjectsClient = () => {
   return savedObjectsClient;
 };
 
-export const loadIndexPatterns = async () => {
-  const { savedObjects } = await getSavedObjectsClient().find({
+export const loadIndexPatterns = async (pattern: string) => {
+  let allSavedObjects = [];
+  const formattedPattern = formatPattern(pattern);
+  const perPage = 1000;
+
+  const { savedObjects, total } = await getSavedObjectsClient().find({
     type: 'index-pattern',
     fields: ['title'],
-    perPage: 10000,
+    page: 1,
+    search: formattedPattern,
+    perPage,
   });
-  return savedObjects;
+
+  allSavedObjects = savedObjects;
+
+  if (total > perPage) {
```

Comment:
We make an initial request with 1000 perPage limitation.
If the total in the response is higher than 1000 we fetch the rest as chunks that are consist of max 1000 items.